### PR TITLE
Merge samza-runner to master

### PR DESCRIPTION
**Pull request by xinyuiscool**
**Please** add a meaningful description for your change here

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

### CODE REVIEW

File: runners/samza/build.gradle [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import groovy.json.JsonOutput
+
+apply from: project(":").file("build_rules.gradle")
```
Comment (kennknowles) [inline runners/samza/build.gradle]: Sorry for the churn, but you'll need to change this to `apply plugin:org.apache.beam.gradle.BeamModulePlugin`

**Comment by xinyuiscool** [comment]
hmm, seems something wrong in the merge. Let me redo it.

File: runners/samza/build.gradle (hunk 1 above)
Comment (xinyuiscool) [reply runners/samza/build.gradle]: Updated the build.gradle with this plugin. Please take a look again. Thanks.

**Comment by kennknowles** [comment]
Checked this:

```
$ git log --oneline ^github/master ^github/samza-runner github/pr/5668
c4607917a1 (github/pr/5668) Update build.gradle with Beam plugin
06bbcfb178 Fix a few typos
bdf202a034 Minor fixes for apache headers and style checks
2e4be30573 Add build.gradle and fix findbugs issues
bbbf659679 Update Beam snapshot version to 2.6.0
c1f2059df1 Add missing apache headers
ac4c64005c [BEAM-3079] Update samza-runner with more features and improvements
4ed3a81715 Add license header to the tests
ab93dc3fe2 Turn on validation tests
72ed480a52 Remove the offsets calculation/comparison in UnboundedSourceSystem admin
f3ad18cca2 Populate beam job name to Samza
017a70652a BEAM-3079: Samza runner
```

This is a mix of:
 - a big squash of most of the developments on `samza-runner`
 - some rewriting of commits that are on the branch so their hashes don't match
 - some fixups just for the merge to master

I know there are different styles to commit history but since this is a big merge I'd like to keep it canonical. Ideally just a PR from `apache:samza-runner` into `apache:master` but also fine to have a history where the PR is just `apache:samza-runner` with a couple of clear final commits.

**Comment by kennknowles** [comment]
This was done elsewhere.

**Pull request closed without merging**